homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -109,7 +109,8 @@ async def async_step_prompt_options(self, user_input=None):
                         f"{DOMAIN}_{CONF_REMOVE_DEVICE}_{device_id}"
                     )
                     self._device_registry.async_remove_device(entry_id)
-                    devices[event_code] = None
+                    if event_code is not None:
+                        devices[event_code] = None
```

Comment:
this leaves an stale entry for the device. I think we'd want:
```suggestion
                        del devices[event_code]
```

Reply:
Here `devices` is not from config entry data. By writing the key `event_code` to `None` and passing it to `update_config_data` it tells the function to remove the key.